src/UKHO.S100PermitService.Common/Services/ManufacturerKeyService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Options;
+using UKHO.S100PermitService.Common.Clients;
+using UKHO.S100PermitService.Common.Configuration;
+using UKHO.S100PermitService.Common.Events;
+using UKHO.S100PermitService.Common.Exceptions;
+using UKHO.S100PermitService.Common.Providers;
+
+namespace UKHO.S100PermitService.Common.Services
+{
+    public class ManufacturerKeyService : IManufacturerKeyService
+    {
+        private readonly IOptions<ManufacturerKeyConfiguration> _manufacturerKeyvault;
+        private readonly ICacheProvider _cacheProvider;
+        private readonly ISecretClient _secretClient;
+
+        public ManufacturerKeyService(IOptions<ManufacturerKeyConfiguration> manufacturerKeyvault,
+                                        ICacheProvider cacheProvider,
+                                        ISecretClient secretClient)
+        {
+            _manufacturerKeyvault = manufacturerKeyvault ?? throw new ArgumentNullException(nameof(manufacturerKeyvault));
+            _cacheProvider = cacheProvider ?? throw new ArgumentNullException(nameof(cacheProvider));
+            _secretClient = secretClient ?? throw new ArgumentNullException(nameof(secretClient));
+            CacheManufacturerKeys();
+        }
+
+        public void CacheManufacturerKeys()
+        {
+            if(!string.IsNullOrEmpty(_manufacturerKeyvault.Value.ServiceUri))
+            {
+                var secretProperties = _secretClient.GetPropertiesOfSecrets();
+
+                if(!secretProperties.Any())
+                {
+                    throw new PermitServiceException(EventIds.ManufacturerIdNotFoundInCache.ToEventId(), "No Secrets found in Manufacturer Keyvault");
```

Comment:
EventId name can be precise like ManufacturerIdNotFoundInKeyVault

Reply:
Done